syncthing/syncthing
# lib/config: Added option for versioning encrypted folders (fixes #8577)

### Purpose

Versioning is required to protect against data loss. Versioning of encrypted folders is always disabled in the current implementation. This change allows you to enable versioning of encrypted folders.

### Testing

1. Default true option value if missing in config
2. Press Save in advanced gui, check config for new option with true value, restart and check for true in gui
3. Change option to false, save, check config for false, restart, check for false in gui
4. With true value check for disabling versioning for auto/manual accept encrypted folder (with enabled in default folder config)
5. With false value check for enabling versioning for auto/manual accept encrypted folder (with enabled in default folder config)

### Screenshots

![Screenshot from 2022-10-04 13-47-47](https://user-images.githubusercontent.com/2958497/193812196-c8973851-e0fe-4d9f-8f3d-63b6c2e7279a.png)

### Documentation

We need documentation changes, something like:

Encrypted Folder Versioning Disabled

Versioning of encrypted folders is disabled by default. This option allows you to use the default folder versioning for encrypted folders.

## Authorship

Your name and email will be added automatically to the AUTHORS file
based on the commit metadata.
